infra/config-store/src/main/java/com/simprints/infra/config/store/ConfigRepositoryImpl.kt

```diff
@@ -26,32 +26,36 @@ internal class ConfigRepositoryImpl @Inject constructor(
 ) : ConfigRepository {
 
     companion object {
+
         @VisibleForTesting
         const val PRIVACY_NOTICE_FILE = "privacy_notice"
     }
 
     override suspend fun getProject(projectId: String): Project = try {
         localDataSource.getProject()
     } catch (e: NoSuchElementException) {
-        refreshProject(projectId)
+        refreshProject(projectId).first
```

Comment:
I considered it, but in most cases, the result of `getProject()` is used as a destructive declaration, so there is little confusion. This might be the only place first/second are called directly, and it is within the repo itself. 
I can add the struct if the team insists.

Reply:
Agree with Alex - a glance at the code doesn't make it clear what`.first` and `.second` are referring to.
ProjectWithConfig is a little better but still a little weird. A third idea - why not make ProjectConfiguration a property of Project? Semantically it IS a property of the project. Plus, we are calling `refreshProject()`, `getProject()`, etc.

Reply:
> Also, the current configuration is often used separately from the whole project config. That would be something to consider in future.
Making it a property of `Project` won't prevent it from being passed independently. 
But the data migration point is a pain, yes. Can we add it as an ignored property, though?

Reply:
Maybe. However, that could be unnecessary complexity at this point since those data structures are rarely used together in the rest of the app.

Reply:
I would favor keeping the `ProjectWithConfig` in its current way because it does precisely what is expected from it - stores the project with its configuration. Meanwhile, the `Project` and `ProjectConfiguration` objects should remain unchanged since on the business domain level and still can be used separately.